packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -28,8 +29,29 @@ export class AssociationManager {
     };
   }
 
+  public static constrainTagUpdates(portfolio: IPortfolio, product: IProduct, options: TagUpdateConstraintOptions): void {
+    this.validateConstraintDescription(this.prettyPrintAssociation(portfolio, product), options);
```

Comment:
Instead of having to call `this.prettyPrintAssociation(portfolio, product)` every time we call `validateConstraintDescription()`, let's instead make it make a `product` and `portfolio`, and `validateConstraintDescription()` will call `prettyPrintAssociation()`):

```suggestion
    this.validateConstraintDescription(options, portfolio, product);
```